app/models/dataset/DatasetService.scala

```diff
@@ -356,15 +357,17 @@ class DatasetService @Inject()(organizationDAO: OrganizationDAO,
       _ <- Fox.serialCombined(pathInfos)(updateRealPath)
     } yield ()
 
-  def getPathsForDataLayer(datasetId: ObjectId, layerName: String): Fox[List[(DatasetMagInfo, List[DatasetMagInfo])]] =
+  def getPathsForDataLayer(datasetId: ObjectId,
+                           layerName: String): Fox[List[(DataSourceMagInfo, List[DataSourceMagInfo])]] =
```

Comment:
Could youu add a comment explaining the signature of this function? I think it’s not immediately obvious why the return type is this nested list